src/js/components/pipeline/PullRequests.jsx

```diff
@@ -30,12 +30,29 @@ const tableContainerSelector = `#${tableContainerId}`;
 
 export default ({ stage, data, status }) => {
   const [selectValue, setSelectValue] = useState(null)
+  const [selectOptions, setSelectOptions] = useState([{ label: 'All', value: null }])
+
+  const options = useMemo(() => {
+    const getLabel = prLabel(stage)
+    const labels = [...data.prs.reduce((acc, curr) => acc.add(getLabel(curr)), new Set())]
+    return [
+      { label: 'All' },
+      ...labels.map(value => ({ value, label: value }))
+    ]
+  }, [data.prs, stage])
+
+  const isReady = status => status === READY
+
+  // reset filter when changing stage
+  useEffect(() => setSelectValue(null), [stage])
```

Comment:
Is this working as expected? I guess you meant:

```suggestion
  useEffect(() => {
      setSelectValue(null)
  }, [stage])
```
?

Reply:
I think the currently you're returning it to be run as cleanup function, but you're not actually returning a function so React will complain

Reply:
true, for a moment forgot it's an implicit return, fixed